web3/_utils/request.py

```diff
@@ -114,19 +127,31 @@ def make_post_request(
 
 # --- async --- #
 
+
 _async_session_cache = SessionCache(size=20)
 _async_session_cache_lock = threading.Lock()
+_pool = ThreadPoolExecutor(max_workers=1)
```

Comment:
Each thread gets 1 max worker because if more are allowed we eventually get caught in a strange situation where the loop ends up closed. I wasn't able to debug further to understand exactly why that is, may have to do with the generator (asynccontextmanager) but that's essentially why I limited it since this is a pretty reliable way to make it more stable.

Reply:
Got it. Thanks!